File: Backend/src/main/java/com/example/demo/Models/PageModel.java
package com.example.demo.Models;


import com.example.demo.DemoApplication;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.core.JsonProcessingException;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;


public class PageModel {

    private long id;
    private String name;
    private String description;
    private String imgPath;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Date createdAt;
    private UserModel admin;
    private List<UserModel> members;
    private List<PostModel> postModels;


    public PageModel() {
        id = 0;
        name = description = imgPath = "test";
        createdAt = new Date();
        admin = null;
        members = new ArrayList<>();
    }

    public List<PostModel> getPostModels() {
        return this.postModels;
    }

    public void setPostModels(List<PostModel> postModels) {
        this.postModels = postModels;
    }

    public List<UserModel> getMembers() {
        return this.members;
    }

    public void setMembers(List<UserModel> members) {
        this.members = members;
    }

    public UserModel getAdmin() {
        return this.admin;
    }

    public void setAdmin(UserModel admin) {
        this.admin = admin;
    }

    public long getId() {
        return this.id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImgPath() {
        return this.imgPath;
    }

    public void setImgPath(String imgPath) {
        this.imgPath = imgPath;
    }

    public Date getCreatedAt() {
        return this.createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    @Override
    public String toString() {
        try {
            return DemoApplication.objectMapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }
        return "";
    }

}

File: Frontend/lib/Model/PageModel.dart
import 'package:frontend/Model/UserModel.dart';
import 'package:frontend/main.dart';

import '../Global/Global.dart';
import 'PostModel.dart';

class PageModel {
  String? id;
  String? name;
  String? description;
  String? imgPath;
  DateTime? createdAt;
  UserModel? admin;
  List<UserModel>? members;
  List<PostModel>? postModels;

  PageModel(
      {this.id,
      this.name,
      this.admin,
      this.members,
      this.imgPath,
      this.description,
      this.createdAt,
      this.postModels});

  PageModel.fromJson(Map<String, dynamic> json) {
    id = json["id"].toString();
    name = json["name"];
    description = json["description"];
    if (json["createdAt"] != null) {
      createdAt = DateTime.parse(json["createdAt"]);
    }
    if (json["imgPath"] != null) imgPath = json["imgPath"];
    if (json["admin"] != null) admin = UserModel.fromJson(json["admin"]);
    if (json['members'] != null) {
      members = [];
      json['members'].forEach((v) {
        members!.add(UserModel.fromJson(v));
      });
    }
    if (json['postModels'] != null) {
      postModels = [];
      json['postModels'].forEach((v) {
        postModels!.add(PostModel.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['name'] = name;
    data['description'] = description;
    data['createdAt'] = createdAt.toString();
    if (imgPath != null) data['imgPath'] = imgPath;
    if (admin != null) data['admin'] = admin!.toJson();
    if (members != null) {
      data['members'] = members!.map((v) => v.toJson()).toList();
    }
    if (postModels != null) {
      data['postModels'] = postModels!.map((v) => v.toJson()).toList();
    }
    return data;
  }

  get getId => id;

  set setId(String id) => this.id = id;

  get getName => name;

  set setName(pageName) => name = name;

  get getAdmin => admin;

  set setAdmin(admin) => this.admin = admin;

  get getMembers => members;

  set setMembers(members) => this.members = members;

  get getImage => imgPath;

  set setImage(imgPath) => this.imgPath = imgPath;

  get getDescription => description;

  set setDescription(description) => this.description = description;

  get getCreatedAt => createdAt;

  set setCreatedAt(createdAt) => this.createdAt = createdAt;

  get getPostModels => postModels;

  set setPostModels(postModels) => this.postModels = postModels;

  static List<PageModel> pages = [
    PageModel(
        id: "1",
        name: "drama news",
        admin: UserModel.users[2],
        members: UserModel.users,
        imgPath: null),
    PageModel(
        id: "2",
        name: "drama news",
        admin: currentUser,
        members: UserModel.users,
        imgPath: "images/photo.jpg"),
    PageModel(
        id: "3",
        name: "drama news",
        admin: UserModel.users[4],
        members: UserModel.users,
        imgPath: null),
    PageModel(
        id: "4",
        name: "drama news",
        admin: UserModel.users[2],
        members: UserModel.users,
        imgPath: "images/photo.jpg"),
    PageModel(
        id: "5",
        name: "drama news",
        admin: currentUser,
        members: UserModel.users,
        imgPath: "images/photo.jpg"),
    PageModel(
        id: "6",
        name: "drama news",
        admin: UserModel.users[3],
        members: UserModel.users,
        imgPath: "images/photo.jpg"),
    PageModel(
        id: "7",
        name: "drama news",
        admin: UserModel.users[2],
        members: UserModel.users,
        imgPath: "images/photo.jpg"),
  ];
}
